File: ios/RNToolbox/TBBatteryManager.m
//
//  TBBatteryManager.m
//  Toolbox
//
//  Created by Larkin Whitaker on 2/2/17.
//  Copyright © 2017 Facebook. All rights reserved.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridge.h>
#import "React/RCTEventEmitter.h"

@interface RCT_EXTERN_MODULE(TBBatteryManager, NSObject)

RCT_EXTERN_METHOD(getBatteryInfo:(RCTResponseSenderBlock)success);
RCT_EXTERN_METHOD(ignore);

@end

File: ios/RNToolbox/TBBatteryManager.swift
//
//  TBBattery.swift
//  TBBatteryMonitor
//
//  Created by Larkin Whitaker on 2/2/17.
//  Copyright © 2017 Facebook. All rights reserved.
//

import Foundation


@objc(TBBatteryManager)
class TBBatteryManager: RCTEventEmitter
{
    override func supportedEvents() -> [String]!
    {
        return ["BatteryStatusChange"]
    }
    
    override init()
    {
        super.init()
        
        Toolbox.Device.current.battery.startMonitoring{_ in
            self.batteryLevelChanged()
        }
    }
    
    @objc(getBatteryInfo:)
    func getBatteryInfo(getBatteryInfo: @escaping RCTResponseSenderBlock)
    {
        getBatteryInfo([self.getBatteryStatus()])
    }
    
    func getBatteryStatus() -> NSDictionary
    {
        return Toolbox.Device.current.battery.info() as NSDictionary
    }
    
    func batteryLevelChanged()
    {
        self.sendEvent(
            withName: "BatteryStatusChange",
            body: self.getBatteryStatus()
        )
    }
    
    @objc(ignore)
    func ignore()
    {
        Toolbox.Device.current.battery.stopMonitoring()
    }
}
